### ISSUE TITLE
Delete referenced contacts

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-core#153]
Deleting a contact does not work under many circumstances that are not always clear to the user - eg. the contact may be used by a message in a group that is out of scope, in the deaddrop, used as cc in a message of a blocked contact and so on.

Contact deletion is mainly used by users to clean up their suggestion list when starting a new chat.

So, if a contact cannot be deleted directly, instead of showing an error, the contact ~~should~~ could be hidden by a decreased origin and all keys and other information should be deleted.
hidden contacts are not new: they are already used now for the deaddrop and other things.

**Comment by testbird** [comment]
What is "the deaddrop"?

**Comment by r10s** [comment]
sorry, it's the internal slang for the contact request :)

**Comment by r10s** [comment]
while updating the profile, i removed the "delete contact" function from there.

- when opening a contact profile, the contact is typically in use, so a deletion will typically fail (and is also often not wanted - blocking is more efficient as this avoids eg. getting further messages from a contact, this cannot be achieved by deletion)

instead, contacts can be deleted by a long-click in the contact list; this is also the place where contacts are created manually, so i think this makes more sense, eg. deleting a contact creating accidentally by hand is no problem.

**Comment by r10s** [comment]
so: maybe this issue is already solved :)

**Comment by csb0730** [comment]
Hi @r10s 
deleting a contact means?
- deleting a chat with a single peer and all of it's messages and attachments locally
- deleting related keys locally
- let messages which are still at server side untouched
- what about group chats where this peer is still in use? Are these chats untouched in this case
or what's the behaviour. My expectation would be that these will exist further on.
- will the address stay in the list of known contacts? Expectation is yes. But not shure what happens in reality.

Maybe it's necessary to distinguish between a contact and a chat? (!)

**Comment by r10s** [comment]
deleting a chat is already possible without problems.
the issue is really about deleting a contact (from the "New chat" list where you also can create one, so it makes sense to delete one that was created accidentially eg. with a spelling error). any yes, deleting a contact requires (at least) all the checks you've mentioned.

**Comment by testbird** [comment]
After deleting all chats and all unapproved emails, there were still some contacts that could not be deleted (other them one's own contact). I wondered, how that could be?

But I also have a question about the blocking used instead of deleting used contacts: Won't that also hide all later messages received from that contact? That's likely not what one wants when deleting a contacts, deleting would not be expected to have such a consequence for the future (other than removing the contact from the current list).

**Comment by testbird** [comment]
Maybe I had just read it wrongly: hiding is not the blocking you wrote about in a later comment.

Deleting: If unused delete completely, else delete info and hide the address from the list (set to disregarded origin)? Receiving a new message will raise the origin and make the address  show up again?

Blocking: A separate contact property?

**Comment by r10s** [comment]
> Hiding is not equal to the blocking

No.

The idea of _hiding_ would be a workaround if we cannot delete a contact.  
If a new message of the hidden/deleted comes in, it would be show up, chat may be created etc. just as if this is the very first message of the contact.

_Blocking_ avoids new messages of a contact to show up, create a chat etc.
To achieve this, of course, we cannot delete the record (we have to know the user does not want to see messages from this user).

However, as we allow deletion only from the contact list (see above) i do no longer feel this is an issue. Moreover, an additional _hiding_ state would make things even more complicated and maybe hard to understandable to the user: eg. he might find the originally mentioned _groups out of scope_ and see the contact in contact list - but he cannot create a new chat with em. Or we would hide the contact there - what with a message sent from the hidden contact to the group? There seems to be far too many complications with a hidden state, so I would tend to close the issue.

**Comment by testbird** [comment]
Ok, good.

Only issue remaining then may be that some contacts could not be deleted, even though the chat list and unapproved emails list was empty.

Possibly the "delete referenced contacts" function could ask to also delete all referencing messages, to get rid of some contact and all its associated things. (Something maybe to be done when polishing edges, after broad usablity is reached.)

NB: Deinstallaton was for reseting (large data usage and delta chat not sending/receiving anything).

**Comment by r10s** [comment]
ok, fine, i think we close this for now. 
nb: the upcoming version shows the chats shared with a contact in the contact's profile - this also makes thinks much clearer for the user.

**Comment by testbird** [comment]
Listing the chats sounds good, it's nice and gives an overview :)

NB: I did not have the chance to get a log, but can pass on the observation that when K-9 is installed in parallel it is usefull to see when DC lags behind.

**Comment by r10s** [comment]
great :)
